FW takeoff docs improvements (#2232)
PX4/PX4-user_guide: en/flight_modes/takeoff.md

<span id="MIS_TAKEOFF_ALT"></span>[MIS_TAKEOFF_ALT](../advanced_config/parameter_reference.md#MIS_TAKEOFF_ALT) | Target altitude during takeoff (default: 2.5m)
<span id="MPC_TKO_SPEED"></span>[MPC_TKO_SPEED](../advanced_config/parameter_reference.md#MPC_TKO_SPEED) | Speed of ascent (default: 1.5m/s)
<span id="COM_RC_OVERRIDE"></span>[COM_RC_OVERRIDE](../advanced_config/parameter_reference.md#COM_RC_OVERRIDE) | Controls whether stick movement on a multicopter (or VTOL in MC mode) causes a mode change to [Position mode](../flight_modes/position_mc.md). This can be separately enabled for auto modes and for offboard mode, and is enabled in auto modes by default.
<span id="COM_RC_STICK_OV"></span>[COM_RC_STICK_OV](../advanced_config/parameter_reference.md#COM_RC_STICK_OV) | The amount of stick movement that causes a transition to [Position mode](../flight_modes/position_mc.md) (if [COM_RC_OVERRIDE](#COM_RC_OVERRIDE) is enabled)

<a id="fixed_wing"></a>
## Fixed-wing (FW)

> Automatic takeoff has two modalities: *catapult/hand-launch* or *runway takeoff*.

Reviewer: @sfuhrer I'm going to merge this now - I think it is OK.

My main concern is that we are mixing the takeoff mode and mission mode takeoff, as discussed. I think we could do a better job here, but maybe as a post process.

What we could do is:
1. Duplicate just this section in mission mode FW takeoff, and make it mission mode specific. 
2. Then reference the catapult and runway sections in takeoff mode for the "common stuff".
3. Remove the mission mode specific stuff from here.

We might then put the stuff currently in mission mode into the "Planning" doc. 

Let me know what you think?

Author: So where would the catapult and runway sections live? In the mission or the flight modes docs, or on a new, neutral ground?

The mode defaults to catapult/hand launch, but can be set to runway takeoff by setting [RWTO_TKOFF](#RWTO_TKOFF) to 1.

There are two ways to start an automatic takeoff on fixed-wing vehicles: either by [planning a mission takeoff](../flight_modes/mission.md#fw-takeoff) and starting the mission, or by switching to the _Takeoff mode_ and arming the vehicle.

In both cases, a flight path (starting point and takeoff course) and clearance altitude are defined.
The flight path takes the vehicle's current position as starting point when the takeoff mode is first entered, and a straight line from this starting point continues in the direction of the defined course indefinitely.
On takeoff, the aircraft will follow this line, climbing at the maximum climb rate ([FW_T_CLMB_MAX](../advanced_config/parameter_reference.md#FW_T_CLMB_MAX)) until reaching the clearance altitude.

In _Takeoff mode_ (non-mission takeoffs), the course is set to the vehicle heading on arming, and the clearance altitude is set to [MIS_TAKEOFF_ALT](#MIS_TAKEOFF_ALT).
Reaching the clearance altitude causes the vehicle to enter _Hold mode_.

> In [Mission mode](../flight_modes/mission.md) the operator defines the takeoff course and clearance altitude in the Takeoff mission item.

Reviewer: FYI So here you see the mission mode all separated into own paragraph.

<span id="COM_RC_STICK_OV"></span>[COM_RC_STICK_OV](../advanced_config/parameter_reference.md#COM_RC_STICK_OV) | The amount of stick movement that causes a transition to [Position mode](../flight_modes/position_mc.md) (if [COM_RC_OVERRIDE](#COM_RC_OVERRIDE) is enabled)

<span id="fixed_wing"></span>
## Fixed-wing (FW)

Automatic takeoff has two modalities: *catapult/hand-launch* or *runway takeoff*.
The mode defaults to catapult/hand launch, but can be set to runway takeoff by setting [RWTO_TKOFF](#RWTO_TKOFF) to 1.

There are two ways to start an automatic takeoff on fixed-wing vehicles: either by [planning a mission takeoff](../flight_modes/mission.md#takeoff) and starting the mission, or by switching to the _Takeoff mode_ and arming the vehicle.

In both cases, a flight path (starting point and takeoff course) and clearance altitude are defined.
The flight path takes the vehicle's current position as starting point when the takeoff mode is first entered, and a straight line from this starting point continues in the direction of the defined course indefinitely.

> On takeoff, the aircraft will follow this line, climbing at the maximum climb rate ([FW_T_CLMB_MAX](../advanced_config/parameter_reference.md#FW_T_CLMB_MAX)) until reaching the clearance altitude.

Reviewer: For a takeoff mode the heading comes from the launch direction. 

For mission mode I'm not certain. What I think happens is that the [TAKEOFF](https://mavlink.io/en/messages/common.html#MAV_CMD_NAV_TAKEOFF) item defines the lat, lon, altitude. 
- Is the altitude in the item the clearance altitude? 
- If you mission launch you'll initially be thrown or head off along runway. That direction might not match the mission item lat/lon. At what point do you head towards that lat/lon?

Author: - Yes that's the clearance altitude.
- it will immediately go towards the mission Takeoff WP lat/lon (keep bearing from start of takeoff to Takeoff WP until past clearance altitude).

Reviewer: No need to answer ^^^. What I've done here is more clearly separated the mission mode and takeoff mode differences. Just a paragraph reorg.

<span id="COM_RC_STICK_OV"></span>[COM_RC_STICK_OV](../advanced_config/parameter_reference.md#COM_RC_STICK_OV) | The amount of stick movement that causes a transition to [Position mode](../flight_modes/position_mc.md) (if [COM_RC_OVERRIDE](#COM_RC_OVERRIDE) is enabled).

<span id="fixed_wing"></span>
## Fixed-wing (FW)

The aircraft takes off in the current direction (heading) using either *catapult/hand-launch mode* or *runway takeoff mode*.
The mode defaults to catapult/hand launch, but can be set to runway takeoff by setting [RWTO_TKOFF](#RWTO_TKOFF) to 1.

> There are generally two ways to start an automatic takeoff on fixed-wing vehicles: either by planning a mission takeoff and starting the mission, or by switching to the Takeoff flight mode and arm the vehicle. Both ways lead to the same behavior, with the difference that in a mission takeoff, the user can specify the takeoff course (vehicle will track this course after being clear of the runway) and takeoff altitude (vehicle will climb to this altitude while staying on the takeoff course). For the non-mission takeoff, the course is set to the vehicle heading on arming, and the altitude is set to [MIS_TAKEOFF_ALT](#MIS_TAKEOFF_ALT).

Reviewer: Layout to my taste
```suggestion
There are generally two ways to start an automatic takeoff on fixed-wing vehicles: either by planning a mission takeoff and starting the mission, or by switching to the Takeoff flight mode and arm the vehicle.
Both methods lead to very similar behavior, with the difference being that in a mission takeoff, the user can specify the takeoff course (vehicle will track this course after being clear of the runway) and takeoff altitude (vehicle will climb to this altitude while staying on the takeoff course).
For the non-mission takeoff, the course is set to the vehicle heading on arming, and the altitude is set to [MIS_TAKEOFF_ALT](#MIS_TAKEOFF_ALT).

```

<span id="COM_RC_STICK_OV"></span>[COM_RC_STICK_OV](../advanced_config/parameter_reference.md#COM_RC_STICK_OV) | The amount of stick movement that causes a transition to [Position mode](../flight_modes/position_mc.md) (if [COM_RC_OVERRIDE](#COM_RC_OVERRIDE) is enabled).

<span id="fixed_wing"></span>
## Fixed-wing (FW)

The aircraft takes off in the current direction (heading) using either *catapult/hand-launch mode* or *runway takeoff mode*.
The mode defaults to catapult/hand launch, but can be set to runway takeoff by setting [RWTO_TKOFF](#RWTO_TKOFF) to 1.

There are generally two ways to start an automatic takeoff on fixed-wing vehicles: either by planning a mission takeoff and starting the mission, or by switching to the Takeoff flight mode and arm the vehicle.
Both methods lead to very similar behavior, with the difference being that in a mission takeoff, the user can specify the takeoff course (vehicle will track this course after being clear of the runway) and takeoff altitude (vehicle will climb to this altitude while staying on the takeoff course).

> For the non-mission takeoff, the course is set to the vehicle heading on arming, and the altitude is set to [MIS_TAKEOFF_ALT](#MIS_TAKEOFF_ALT).

Reviewer: Does this mean on reaching the altitude the vehicle will continue forever on the arming-set course in Takeoff mode? Or will it switch to some other mode on reaching the altitude (e.g. loiter/hold)?

Reviewer: I see below it switches to hold. We should say that at this level.

I also "wonder" if we should pull the mission-mode takeoff information here into a note or its own section. I.e. I think this is useful to state, but we should try keep this whole doc as much about FW takeoff mode as possible and therefore keep the mission mode comparisons as an aside.

Author: >Does this mean on reaching the altitude the vehicle will continue forever on the arming-set course in Takeoff mode? Or will it switch to some other mode on reaching the altitude (e.g. loiter/hold)?

It will switch to Hold and circle around Home (Takeoff without mission), proceed with the mission otherwise

Author: Yes please feel free to add the info about Hold here.

Separating all the Mission-takeoff related things to a note sounds reasonable to me indeed. We anyway should add some very rough bullet points on how to plan a Mission Takeoff also to the Mission section, and then reference to the docs here (like "place a Takeoff item in the direction of the runway, and set the altitude higher than obstacles in close proximity".

<span id="FW_TKO_AIRSPD"></span>[FW_TKO_AIRSPD](../advanced_config/parameter_reference.md#FW_TKO_AIRSPD) | Takeoff airspeed (is set to [FW_AIRSPD_MIN](../advanced_config/parameter_reference.md#FW_AIRSPD_MIN) if not defined by operator)
<span id="FW_TKO_PITCH_MIN"></span>[FW_TKO_PITCH_MIN](../advanced_config/parameter_reference.md#FW_TKO_PITCH_MIN) | This is the minimum pitch angle setpoint during the climbout phase.
<span id="FW_THR_MAX"></span>[FW_THR_MAX](../advanced_config/parameter_reference.md#FW_THR_MAX) | Maximum throttle during the whole takeoff phase.
<span id="FW_T_CLMB_MAX"></span>[FW_T_CLMB_MAX](../advanced_config/parameter_reference.md#FW_T_CLMB_MAX) | Climb rate setpoint during climbout phase.

<span id="hand_launch"></span>
### Catapult/Hand Launch

In *catapult/hand-launch mode* the vehicle waits to detect launch (based on acceleration trigger).
On launch it ramps up to full throttle [FW_THR_MAX](#FW_THR_MAX) and climbs with the maximum climb rate [FW_T_CLMB_MAX](#FW_T_CLMB_MAX) while keeping the pitch setpoint above [FW_TKO_PITCH_MIN](#FW_TKO_PITCH_MIN).

> Once it reaches [MIS_TAKEOFF_ALT](#MIS_TAKEOFF_ALT) (or altitude defined in the mission item) it will automatically switch to [Hold mode](../flight_modes/hold.md) and loiter.

Reviewer: This is the takeoff mode doc - if you have that caveat this is incorrect because in a mission it will continue to the next mission item before continuing.
I think if you can have a catapult launch in a mission mode, we need to have that information in the mission mode doc or "below" it. 

```suggestion
Once it reaches [MIS_TAKEOFF_ALT](#MIS_TAKEOFF_ALT) it will automatically switch to [Hold mode](../flight_modes/hold.md) and loiter.

```

Author: Yes this should be cleanly separated: Loiter in case of mission-less takeoff, proceed to next WP in Mission.

Reviewer 2: there's also some repeat info in the paragraph from before and this one. maybe could do another revision pass through and even just describe each mode separately (with less general description beforehand)

Author: removed reference to the mission item altitude here.

All RC stick movement is ignored during the full takeoff sequence.

To launch in this mode:
- Arm the vehicle.

> - Put the vehicle into *Takeoff mode* or *Start mission with current waypoint being a Takeoff*

Reviewer: Again, this is the takeoff mode doc. 

```suggestion
- Put the vehicle into *Takeoff mode*

```

Author: removed it.

- Launch/throw the vehicle (firmly) directly into the wind. You can also shake the vehicle first, wait till the motor spins up and throw only then.

The Launch detector is affected by the following parameters:

Parameter | Description
--- | ---
<span id="LAUN_ALL_ON"></span>[LAUN_ALL_ON](../advanced_config/parameter_reference.md#LAUN_ALL_ON) | Enable automatic launch detection. If disabled motors spin up on arming already.
<span id="FW_LAUN_AC_THLD"></span>[FW_LAUN_AC_THLD](../advanced_config/parameter_reference.md#FW_LAUN_AC_THLD) | Acceleration threshold (acceleration in body-forward direction must be above this value).
<span id="FW_LAUN_AC_T"></span>[FW_LAUN_AC_T](../advanced_config/parameter_reference.md#FW_LAUN_AC_T) | Trigger time (acceleration must be above threshold for this amount of seconds).
<span id="FW_LAUN_MOT_DEL"></span>[FW_LAUN_MOT_DEL](../advanced_config/parameter_reference.md#FW_LAUN_MOT_DEL) | Delay from launch detection to motor spin up.

<span id="runway_launch"></span>
### Runway Takeoff

> For vehicles with a landing gear and steerable wheel. Enable wheel controller via [FW_W_EN](#FW_W_EN). Vehicle should be centered and aligned with runway when takeoff is initiated (either by arming in takeoff mode or by starting a mission with a takeoff mission item as active waypoint). The operator can "nudge" the vehicle while on the runway to help keeping it centered and aligned (see [RWTO_NUDGE](../advanced_config/parameter_reference.md#RWTO_NUDGE)).

Reviewer: ```suggestion
For vehicles with a landing gear and steerable wheel. Enable wheel controller via [FW_W_EN](#FW_W_EN).
Vehicle should be centered and aligned with runway when takeoff is initiated (either by arming in takeoff mode or by starting a mission with a takeoff mission item as active waypoint).
The operator can "nudge" the vehicle while on the runway to help keeping it centered and aligned (see [RWTO_NUDGE](../advanced_config/parameter_reference.md#RWTO_NUDGE)).

```

To launch in this mode:
- Arm the vehicle.
- Put the vehicle into *Takeoff mode* or *Start mission with current waypoint being a Takeoff*
- Launch/throw the vehicle (firmly) directly into the wind. You can also shake the vehicle first, wait till the motor spins up and throw only then.

The Launch detector is affected by the following parameters:

Parameter | Description
--- | ---
<span id="LAUN_ALL_ON"></span>[LAUN_ALL_ON](../advanced_config/parameter_reference.md#LAUN_ALL_ON) | Enable automatic launch detection. If disabled motors spin up on arming already.
<span id="FW_LAUN_AC_THLD"></span>[FW_LAUN_AC_THLD](../advanced_config/parameter_reference.md#FW_LAUN_AC_THLD) | Acceleration threshold (acceleration in body-forward direction must be above this value).
<span id="FW_LAUN_AC_T"></span>[FW_LAUN_AC_T](../advanced_config/parameter_reference.md#FW_LAUN_AC_T) | Trigger time (acceleration must be above threshold for this amount of seconds).
<span id="FW_LAUN_MOT_DEL"></span>[FW_LAUN_MOT_DEL](../advanced_config/parameter_reference.md#FW_LAUN_MOT_DEL) | Delay from launch detection to motor spin up.

<span id="runway_launch"></span>
### Runway Takeoff

> For vehicles with a landing gear and steerable wheel. Enable wheel controller via [FW_W_EN](#FW_W_EN).

Reviewer: ```suggestion
For vehicles with a landing gear and steerable wheel.
Enable wheel controller via [FW_W_EN](#FW_W_EN).

```

To launch in this mode:
- Arm the vehicle.
- Put the vehicle into *Takeoff mode* or *Start mission with current waypoint being a Takeoff*
- Launch/throw the vehicle (firmly) directly into the wind. You can also shake the vehicle first, wait till the motor spins up and throw only then.

The Launch detector is affected by the following parameters:

Parameter | Description
--- | ---
<span id="LAUN_ALL_ON"></span>[LAUN_ALL_ON](../advanced_config/parameter_reference.md#LAUN_ALL_ON) | Enable automatic launch detection. If disabled motors spin up on arming already.
<span id="FW_LAUN_AC_THLD"></span>[FW_LAUN_AC_THLD](../advanced_config/parameter_reference.md#FW_LAUN_AC_THLD) | Acceleration threshold (acceleration in body-forward direction must be above this value).
<span id="FW_LAUN_AC_T"></span>[FW_LAUN_AC_T](../advanced_config/parameter_reference.md#FW_LAUN_AC_T) | Trigger time (acceleration must be above threshold for this amount of seconds).
<span id="FW_LAUN_MOT_DEL"></span>[FW_LAUN_MOT_DEL](../advanced_config/parameter_reference.md#FW_LAUN_MOT_DEL) | Delay from launch detection to motor spin up.

<span id="runway_launch"></span>
### Runway Takeoff

> For vehicles with a landing gear and steerable wheel.
> Enable wheel controller via [FW_W_EN](#FW_W_EN).

Reviewer: I think you are saying...
```suggestion
Runway takeoffs can be used by vehicles with landing gear and and steerable wheel (only).
You will first need to enable the wheel controller using the parameter [FW_W_EN](#FW_W_EN).

```

The operator can "nudge" the vehicle while on the runway to help keeping it centered and aligned (see [RWTO_NUDGE](../advanced_config/parameter_reference.md#RWTO_NUDGE)).

The *runway takeoff mode* has the following phases:
1. **Throttle ramp**: Throttle is ramped up within [RWTO_RAMP_TIME](../advanced_config/parameter_reference.md#RWTO_RAMP_TIME) to [RWTO_MAX_THR](../advanced_config/parameter_reference.md#RWTO_MAX_THR).
1. **Clamped to runway**: Pitch fixed, no roll and course hold until the rotation airspeed ([RWTO_ROT_AIRSPD](../advanced_config/parameter_reference.md#RWTO_ROT_AIRSPD)) is reached. The operator is able to nudge the vehicle left/right via yaw stick.
1. **Climbout**: Increase pitch setpoint and climb to takeoff altitude. To prevent wingstrikes, the controller will keep the roll setpoint locked to 0 when close to the ground, and then gradually allow more roll while climbing. It is based on the vehicle geometry as configured in [FW_WING_SPAN](#FW_WING_SPAN) and [FW_WING_HEIGHT](#FW_WING_HEIGHT).

:::note

> For a smooth takeoff, the runway wheel controller possibly needs to be tuned. It consists of a rate controller (P-I-FF-controller with the parameters [FW_WR_P](../advanced_config/parameter_reference.md#FW_WR_P), [FW_WR_I](../advanced_config/parameter_reference.md#FW_WR_I), [FW_WR_FF](../advanced_config/parameter_reference.md#FW_WR_FF)) and an outerloop that calculates heading setpoints from course errors and can be tuned via [RWTO_L1_PERIOD](#RWTO_L1_PERIOD).

Reviewer: ```suggestion
For a smooth takeoff, the runway wheel controller possibly needs to be tuned.
It consists of a rate controller (P-I-FF-controller with the parameters [FW_WR_P](../advanced_config/parameter_reference.md#FW_WR_P), [FW_WR_I](../advanced_config/parameter_reference.md#FW_WR_I), [FW_WR_FF](../advanced_config/parameter_reference.md#FW_WR_FF)) and an outer loop that calculates heading setpoints from course errors and can be tuned via [RWTO_L1_PERIOD](#RWTO_L1_PERIOD).

```

For the non-mission takeoff, the course is set to the vehicle heading on arming, and the altitude is set to [MIS_TAKEOFF_ALT](#MIS_TAKEOFF_ALT).

Parameters that apply to both catapult/hand-launch as well as runway takeoffs:

Parameter | Description
--- | ---
<span id="MIS_TAKEOFF_ALT"></span>[MIS_TAKEOFF_ALT](../advanced_config/parameter_reference.md#MIS_TAKEOFF_ALT) | Altitude setpoint above Home that the vehicle will climb to during takeoff if not in mission mode.
<span id="FW_TKO_AIRSPD"></span>[FW_TKO_AIRSPD](../advanced_config/parameter_reference.md#FW_TKO_AIRSPD) | Takeoff airspeed (is set to [FW_AIRSPD_MIN](../advanced_config/parameter_reference.md#FW_AIRSPD_MIN) if not defined by operator)
<span id="FW_TKO_PITCH_MIN"></span>[FW_TKO_PITCH_MIN](../advanced_config/parameter_reference.md#FW_TKO_PITCH_MIN) | This is the minimum pitch angle setpoint during the climbout phase.
<span id="FW_THR_MAX"></span>[FW_THR_MAX](../advanced_config/parameter_reference.md#FW_THR_MAX) | Maximum throttle during the whole takeoff phase.

> <span id="FW_T_CLMB_MAX"></span>[FW_T_CLMB_MAX](../advanced_config/parameter_reference.md#FW_T_CLMB_MAX) | Climb rate setpoint during climbout phase.

Reviewer 2: maybe good to indicate that this SHOULD NOT be changed specifically for a different desired climb rate, as this is a tuning parameter for TECS. -- or just mention it as the value that is used in paragraph, but then only list in the table here "configurable" parameters for adjusting the takeoff behavior?

Author: Should we just remove it from the list here? As you're right, the last thing that should happen is that users tune this param for a specific takeoff climb rate. 
It's already mentioned above in the text that the vehicle will climb with this value, so listing it here as a "core parameter to tune for takeoff" is really not necessary.

Author: Done in 023be07d0c

<span id="hand_launch"></span>
### Catapult/Hand Launch

In *catapult/hand-launch mode* the vehicle waits to detect launch (based on acceleration trigger).

> On launch it ramps up to full throttle [FW_THR_MAX](#FW_THR_MAX) and climbs with the maximum climb rate [FW_T_CLMB_MAX](#FW_T_CLMB_MAX) while keeping the pitch setpoint above [FW_TKO_PITCH_MIN](#FW_TKO_PITCH_MIN).

Reviewer 2: No ramp on catapult/hand launch. Also delegates throttle command to TECS, no explicit setting of max throttle.

https://github.com/PX4/PX4-Autopilot/blob/00bfd5436a6f743fe79d6c300ab37982fcf2987f/src/modules/fw_pos_control_l1/FixedwingPositionControl.cpp#L1538-L1583

Author: True, removed it.

<span id="FW_LAUN_AC_T"></span>[FW_LAUN_AC_T](../advanced_config/parameter_reference.md#FW_LAUN_AC_T) | Trigger time (acceleration must be above threshold for this amount of seconds).
<span id="FW_LAUN_MOT_DEL"></span>[FW_LAUN_MOT_DEL](../advanced_config/parameter_reference.md#FW_LAUN_MOT_DEL) | Delay from launch detection to motor spin up.

<span id="runway_launch"></span>
### Runway Takeoff

Runway takeoffs can be used by vehicles with landing gear and and steerable wheel (only).
You will first need to enable the wheel controller using the parameter [FW_W_EN](#FW_W_EN).

Vehicle should be centered and aligned with runway when takeoff is initiated (either by arming in takeoff mode or by starting a mission with a takeoff mission item as active waypoint).
The operator can "nudge" the vehicle while on the runway to help keeping it centered and aligned (see [RWTO_NUDGE](../advanced_config/parameter_reference.md#RWTO_NUDGE)).

The *runway takeoff mode* has the following phases:
1. **Throttle ramp**: Throttle is ramped up within [RWTO_RAMP_TIME](../advanced_config/parameter_reference.md#RWTO_RAMP_TIME) to [RWTO_MAX_THR](../advanced_config/parameter_reference.md#RWTO_MAX_THR).

> 1. **Clamped to runway**: Pitch fixed, no roll and course hold until the rotation airspeed ([RWTO_ROT_AIRSPD](../advanced_config/parameter_reference.md#RWTO_ROT_AIRSPD)) is reached. The operator is able to nudge the vehicle left/right via yaw stick.

Reviewer 2: not sure about calling it "course hold" - as it is actually following a line. maybe "controls to takeoff path"

> 1. **Clamped to runway**: Pitch fixed, no roll and course hold until the rotation airspeed ([RWTO_ROT_AIRSPD](../advanced_config/parameter_reference.md#RWTO_ROT_AIRSPD)) is reached. The operator is able to nudge the vehicle left/right via yaw stick.

Author: Magically the rendering was correct (1,2,3) even with all 1. in the raw:) Anyway changed it now.

It consists of a rate controller (P-I-FF-controller with the parameters [FW_WR_P](../advanced_config/parameter_reference.md#FW_WR_P), [FW_WR_I](../advanced_config/parameter_reference.md#FW_WR_I), [FW_WR_FF](../advanced_config/parameter_reference.md#FW_WR_FF)) and an outer loop that calculates heading setpoints from course errors and can be tuned via [RWTO_L1_PERIOD](#RWTO_L1_PERIOD).

:::

Runway takeoff important parameters:

Parameter | Description
--- | ---
<span id="RWTO_TKOFF"></span>[RWTO_TKOFF](../advanced_config/parameter_reference.md#RWTO_TKOFF) | Enable runway takeoff.
<span id="FW_W_EN"></span>[FW_W_EN](../advanced_config/parameter_reference.md#FW_W_EN) | Enable wheel controller.
<span id="RWTO_MAX_THR"></span>[RWTO_MAX_THR](../advanced_config/parameter_reference.md#RWTO_MAX_THR) | Max throttle during runway takeoff.
<span id="RWTO_RAMP_TIME"></span>[RWTO_RAMP_TIME](../advanced_config/parameter_reference.md#RWTO_RAMP_TIME) | Throttle ramp up time.
<span id="RWTO_ROT_AIRSPD"></span>[RWTO_ROT_AIRSPD](../advanced_config/parameter_reference.md#RWTO_ROT_AIRSPD) | Airspeed threshold to start rotation (pitching up). If not configured by operator is set to 0.9*FW_TKO_AIRSPD.

> <span id="FW_TKO_AIRSPD"></span>[FW_TKO_AIRSPD](../advanced_config/parameter_reference.md#FW_TKO_AIRSPD) | Airspeed setpoint during takeoff. If not configured by operator is set to FW_AIRSPD_MIN.

Reviewer 2: ```suggestion
<span id="FW_TKO_AIRSPD"></span>[FW_TKO_AIRSPD](../advanced_config/parameter_reference.md#FW_TKO_AIRSPD) | Airspeed setpoint during the takeoff climbout phase (after rotation). If not configured by operator is set to FW_AIRSPD_MIN.

```
